### PR TITLE
22 - Review of main readme and model_save_run

### PR DESCRIPTION
**Pull request by izzyStewart**
Hi Jyl and Pius,

This pull request can be broken into two parts to review:

1. Review the main `readme.md` and run `data_splitting_survey.py` in `pipeline/classfication_model/` to produce the training, validation and test datasets.

2. Review the code to train and run the final x2 models found in `cci_nepal/pipeline/classification_model/model_save_run.` This relies on the training and test datasets created by running step 1. The outputs are x2 prediction files for the basic and non-basic models. The `model_save_run folder` has a README as well to guide.

---

Checklist:

- [x ] I have refactored my code out from `notebooks/`
- [ x] I have checked the code runs
- [ x] I have tested the code
- [x ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ x] I have merged any new changes from `dev`
- [ x] I have documented the code
  - [ x] Major functions have docstrings
  - [ x] Appropriate information has been added to `README`s
- [ x] I have explained the feature in this PR or (better) in `output/reports/`
- [ x] I have requested a code review

### CODE REVIEW

File: cci_nepal/pipeline/classification_model/model_save_run/model_run.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+from sklearn.compose import ColumnTransformer
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import RobustScaler
+from sklearn.multioutput import MultiOutputClassifier
+from sklearn.linear_model import LogisticRegression
+import pickle
+
+# Project libraries
+import cci_nepal
+from cci_nepal.getters.classification_model import get_real_data as grd
```
Comment (djyldyz) [inline cci_nepal/pipeline/classification_model/model_save_run/model_run.py:43]: As folders were renamed, this needs to be `from cci_nepal.getters.classification_model import get_data as grd`

File: cci_nepal/pipeline/classification_model/model_save_run/model_save.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+# Fit model
+basic_model = logr_b.fit(X_train_basic, y_train_basic)
+# Save model to disk
+filename = f"{project_dir}/outputs/models/final_classification_model_basic.sav"
+pickle.dump(basic_model, open(filename, "wb"))
+
+# Fit model
+non_basic_model = logr_nb.fit(X_train_non_basic, y_train_non_basic)
+# Save model to disk
+filename = f"{project_dir}/outputs/models/final_classification_model_non_basic.sav"
```
Comment (djyldyz) [inline cci_nepal/pipeline/classification_model/model_save_run/model_save.py:138]: This failed for me as I didn't have a nested folder `models`. Can we try to check for existing folder and create folder if necessary?

**Review by djyldyz — COMMENTED** [review]
Hi both! I'm glad to report that with your help and small tweaks I was able to run the model. I've added comments to files where small changes are necessary to make scripts run.

I was hoping to have a closer look at the code over the next couple of weeks. Are there any specific scripts you want me to go through?

File: cci_nepal/pipeline/classification_model/data_splitting_survey.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+import cci_nepal
+import pandas as pd
+import numpy as np
+from sklearn.model_selection import train_test_split
+import warnings
+from sklearn.utils import shuffle
+
+warnings.filterwarnings("ignore")
+from cci_nepal.getters.real_data import get_real_data as grd
```
[the remark this answers is not in the document]
Comment (izzyStewart) [reply cci_nepal/pipeline/classification_model/data_splitting_survey.py]: Updated thanks!

File: cci_nepal/pipeline/classification_model/data_splitting_survey.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+# Set project directory
+project_dir = cci_nepal.PROJECT_DIR
+
+# Read in df
+df = grd.read_csv_file(f"{project_dir}/inputs/data/real_data/Full_Data_District.csv")
```
[the remark this answers is not in the document]
Comment (izzyStewart) [reply cci_nepal/pipeline/classification_model/data_splitting_survey.py]: Good point and also covers off those without access to the s3 bucket. I think it is like we discussed previously we need to have a dummy dataset and show the results of the model on the dummy dataset (as an alternative to them building the model)

@saurav-datageek FYI

File: cci_nepal/pipeline/classification_model/data_splitting_survey.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+train = pd.concat([train_hill, train_terai], ignore_index=True)
+val = pd.concat([val_hill, val_terai], ignore_index=True)
+
+# Re-shuffle sets
+train = shuffle(train, random_state=1)
+# Re-shuffle sets
+val = shuffle(val, random_state=1)
+
+# Save train and val sets
+train.to_csv(f"{project_dir}/outputs/data/data_for_modelling/train.csv", index=False)
```
[the remark this answers is not in the document]
Comment (izzyStewart) [reply cci_nepal/pipeline/classification_model/data_splitting_survey.py:83]: Thanks for the above suggestion! Have added the line below into the data splitting script.

`Path(f"{project_dir}/outputs/data/data_for_modelling/").mkdir(parents=True, exist_ok=True)`

@saurav-datageek fyi

File: cci_nepal/pipeline/classification_model/model_save_run/model_save.py (hunk 2 above)
Comment (izzyStewart) [reply cci_nepal/pipeline/classification_model/model_save_run/model_save.py:138]: Updated like the above suggestion to add a folder called models if not already created - thanks!

**Pull request merged**